ros/api/common/add_group_filter.py

```diff
@@ -1,12 +1,24 @@
 from ros.lib.models import System
 from flask import request
+from ros.lib.feature_flags import FLAG_INVENTORY_GROUPS, get_flag_value
+from ros.lib.config import get_logger
+
+logger = get_logger(__name__)
 
 
 def group_filtered_query(query):
-    if hasattr(request, "host_groups"):
-        host_groups = [gid for gid in request.host_groups if gid is not None]
-        if len(host_groups) >= 1:
-            query = query.filter(System.groups[0]['id'].astext.in_(host_groups) | (System.groups == '[]'))
+    if get_flag_value(FLAG_INVENTORY_GROUPS):
```

Comment:
@upadhyeammit according to me the feature flag checks should be at the API level as well and for RBAC if feature flag is enabled it should follow the condition based on RBAC response else we should skip adding any filter to the query is what I feel from the functionality point of view so that it will return all the systems and work like before. Let me know your thoughts!

Reply:
What I mean by API level is that if feature flag is disabled that group info should not come from the Systems API response and filtering and sorting on `group_name` should not work.

Reply:
As per new changes if unleash is not enabled then we return empty groups array instead of completely getting rid of groups field from the response. This is to keep in sync with openapi spec